File: V2EX/Nodes/RLNodesVC.m
//
//  RLNodesVC.m
//  V2EX
//
//  Created by ucs on 16/1/5.
//  Copyright © 2016年 ucs. All rights reserved.
//

#import "RLNodesVC.h"
#import "RLBubblesView.h"
#import "RLNode.h"
#import "RLNodeTopicsTVC.h"
#import "MJExtension.h"

@interface RLNodesVC ()
@property (nonatomic, strong) UIScrollView *nodesScrollView;
@property (nonatomic, weak) RLBubblesView *bubblesView;
@end

@implementation RLNodesVC

- (void)viewDidLoad {
    [super viewDidLoad];
    [self loadData];
    
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    self.navigationController.navigationBar.hidden = YES;
}

- (void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
    self.navigationController.navigationBar.hidden = NO;
}

- (void)loadData {
    [[RLNetWorkManager shareRLNetWorkManager] requestWithPath:@"/api/nodes/all.json" success:^(id response) {
        NSArray *AllNodes = [RLNode mj_objectArrayWithKeyValuesArray:response];
        NSMutableArray *nodeModels = [NSMutableArray array];
        for (RLNode *node in AllNodes) {
            if (node.topics.intValue > 100) {
                [nodeModels addObject:node];
            }
        }
        self.bubblesView.nodeModels = nodeModels;
    } failure:^{
    }];
}

#pragma mark ------------------------------------------------------------
#pragma mark 懒加载
- (RLBubblesView *)bubblesView {
    if (!_bubblesView) {
        RLBubblesView *bubblesView = [[RLBubblesView alloc] initWithFrame:[UIScreen mainScreen].bounds];
        bubblesView.clickAction = ^(RLNode *nodeModel){
            RLNodeTopicsTVC *nodeTopicsTVC = [[RLNodeTopicsTVC alloc] init];
            nodeTopicsTVC.nodeModel = nodeModel;
            self.hidesBottomBarWhenPushed = YES;
            [self.navigationController pushViewController:nodeTopicsTVC animated:YES];
            self.hidesBottomBarWhenPushed = NO;
        };
        _bubblesView = bubblesView;
        [self.view addSubview:_bubblesView];
    }
    return _bubblesView;
}

@end

File: V2EX/Nodes/RLNodesVC.swift
//
//  RLNodesVC.swift
//  V2EX
//
//  Created by LLZ on 16/3/24.
//  Copyright © 2016年 ucs. All rights reserved.
//

import UIKit

class RLNodesVC: UIViewController {

    private lazy var bubblesView:RLBubblesView = {
        var bView = RLBubblesView.init(frame: UIScreen.mainScreen().bounds)
        bView.nodeBtnAction = {[weak self] (nodeModel)->Void in
            if let strongSelf = self {
                let nodeTopicsTVC = RLNodeTopicsTVC.init(style: .Grouped)
                nodeTopicsTVC.nodeModel = nodeModel
                strongSelf.hidesBottomBarWhenPushed = true
                strongSelf.navigationController?.pushViewController(nodeTopicsTVC, animated:true)
                strongSelf.hidesBottomBarWhenPushed = false
            }
        }
        return bView
    }()
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.addSubview(self.bubblesView)
        loadData()
        // Do any additional setup after loading the view.
    }
    
    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        self.navigationController?.navigationBar.hidden = true
    }
    
    override func viewWillDisappear(animated: Bool) {
        super.viewWillDisappear(animated)
        self.navigationController?.navigationBar.hidden = false
    }

    private func loadData()  {
        RLNetWorkManager.shareRLNetWorkManager().requestWithPath("/api/nodes/all.json", success: {[weak self] (response) in
            let allNodes = RLNode.mj_objectArrayWithKeyValuesArray(response)
            let nodeModels:NSMutableSet = NSMutableSet.init(array: allNodes as [AnyObject])
            for node in nodeModels{
                if Int((node as! RLNode).topics!)! < 100 {//话题数目少于100条的不显示
                    nodeModels.removeObject(node)
                }
            }
            if let strongSelf = self {
                strongSelf.bubblesView.nodeModels = nodeModels
            }
            }, failure:{})
    }

}
